imports/database/database-access-object.js

```diff
@@ -0,0 +1,37 @@
+export class DatabaseAccessObject {
+    constructor(collection) {
+        this._collection = collection;
+    }
+
+    find(queryObject) {
+        return this._collection.find(queryObject);
+    }
+
+    findOne(queryObject) {
+        return this._collection.findOne(queryObject);
+    }
+
+    insert(object) {
+        this._collection.schema.validate(object);
+        this._collection.insert(object);
+    }
+
+    update(id, object) {
+        let isValid = true;
+        object.keys().forEach(function (key) {
+            isValid &= this._collection.schema.newContext().validateOne(object, key);
```

Comment:
why do you need `newContext()` here? I read the docs but still don't quite understand what it does